BoyumFoosballStats/Pages/PlayerDashboard/Models/PlayerDashboardViewModel.cs

```diff
@@ -3,93 +3,181 @@
+        
+        if (!string.IsNullOrEmpty(PlayerId))
+        {
+            SelectedPlayer = Players.First(x => x.Id == PlayerId);
+            DisplayCharts(PlayerId);
+        }
     }
-    
-    private void DisplayWinRateChart()
+
+    public void HandlePlayerClicked(Player player)
     {
-        var last5WeeksData = GetPlayerWinRateForLast5Weeks();
-        WeekChartData = last5WeeksData.Select(k => new WeekChartDataItem
+        SelectedPlayer = player;
+        DisplayCharts(player.Id);
+    }
+
+    public string GetWinRateToString(Player player)
+    {
+        return $"{GetWinRate(player.Id!, Matches) * 100:0.##}%";
+    }
+
+    private void DisplayCharts(string? playerId)
+    {
+        var winRateData = GetPlayerWinRateForLast5Weeks(playerId!);
+        WinRateByWeekChartData = winRateData.Select(k => new ChartDataItem
+        {
+            XData = k.Key,
+            YData = k.Value
+        }).ToList();
+        
+        var matchesPlayedData = GetMatchesPlayedForLast5Weeks(playerId!);
+        MatchesWeekChartData = matchesPlayedData.Select(k => new ChartDataItem
+        {
+            XData = k.Key,
+            YData = k.Value
+        }).ToList();
+        
+        var winRateByDayData = GetPlayerWinRateByWeekDay(playerId!);
+        WinRateByDayChartData = winRateByDayData.Select(k => new ChartDataItem
         {
-            Date = k.Key,
-            WinRate = k.Value
+            XData = k.Key,
+            YData = k.Value
         }).ToList();
     }
 
-    private Dictionary<string, double> GetPlayerWinRateForLast5Weeks()
+    private Dictionary<string, double> GetPlayerWinRateForLast5Weeks(string playerId)
     {
-        var endDate = DateTime.Today; 
-        var startDate = endDate.AddDays(-35); // Start of the 5th week from the current date
+        var last5WeeksMatches = GetLast5WeekMatches(playerId);
 
-        var relevantMatches = Matches
-            .Where(m => m.MatchDate >= startDate && m.MatchDate <= endDate)
-            .Where(m => 
-                m.BlackAttackerPlayer?.Id == PlayerId || 
-                m.BlackDefenderPlayer?.Id == PlayerId || 
-                m.GreyAttackerPlayer?.Id == PlayerId || 
-                m.GreyDefenderPlayer?.Id == PlayerId).ToList();
+        // Group by week number, calculate win rate for each group, and take the last 5 weeks
+        var winRateByWeek = last5WeeksMatches
+            .GroupBy(m => CultureInfo.CurrentCulture.Calendar.GetWeekOfYear(m.MatchDate, CalendarWeekRule.FirstFullWeek, DayOfWeek.Monday))
+            .OrderBy(g => g.Key)
+            .Take(5)
+            .ToDictionary(
+                g => g.Key.ToString("00"), 
+                g => GetWinRate(playerId, g.ToList())
+            );
 
-        var winRateByWeek = new Dictionary<string, double>();
+        return winRateByWeek;
+    }
 
-        var calendar = CultureInfo.CurrentCulture.Calendar;
+    private Dictionary<string, double> GetPlayerWinRateByWeekDay(string playerId)
```

Comment:
Possibly move to the MatchAnalysisService when the other PR is merged

Reply:
moved to player analysis